### ISSUE TITLE
Remove C++/CX from the codebase

### ISSUE THREAD

**Issue by TheStormN** [microsoft/cpprestsdk#1753]
C++/WinRT is now the recommended way to write UWP apps. Also C++/CX is not compatible with C++20.

I'm trying to migrate our project to C++20 which have dependency on `cpprestsdk`, but unfortunately the linker fails saying the standard libraries are incompatible. I'm getting this error:
```
vccorlib_lib_should_be_specified_before_msvcrt_lib_to_linker': value '1' doesn't match value '0' in msvcrt.lib
```

**Comment by pratikpc** [comment]
@TheStormN was this because your CMake App (if it was CMake) had the [Windows Store](https://cmake.org/cmake/help/latest/variable/WINDOWS_STORE.html) environment variable turned on?

This is present [in their CMakeLists.txt](https://github.com/microsoft/cpprestsdk/blob/411a109150b270f23c8c97fa4ec9a0a4a98cdecf/Release/CMakeLists.txt#L189)

**Comment by TheStormN** [comment]
@pratikpc Yes, we build our software for UWP too. But anyway, In one of the compiler updates, they've fixed the compatibility with C++20 and C++/CX.
